inbox/util/blockstore.py

```diff
@@ -97,10 +150,44 @@ def get_from_blockstore(data_sha256) -> Optional[bytes]:
         log.warning("No data returned!")
         return value
 
+    if check_sha:
+        assert (
+            data_sha256 == sha256(value).hexdigest()
+        ), "Returned data doesn't match stored hash!"
```

Comment:
This is added here because previously we would check hash here, there was no compression involved before. This is still `True` by default but when retrieving MIME we have to decompress it before checking the hash and this is now done in `get_raw_mime` below.